<launch>
    <arg name="robot_name" default="lwr"/>
    <arg name="base_frame"  default="link_0"/>
    <arg name="urdf_url"  default="$(find orca)/examples/resources/lwr.urdf"/>
    <arg name="robot_compensates_gravity" default="true"/>
    <arg name="task_name"  default="CartTask_EE"/>
    <arg name="rviz"  default="true"/>
    <arg name="interactive_marker_frame_id"  default="$(arg base_frame)"/>


    <node name="minimal_controller" pkg="orca_ros" type="minimal_controller" output="screen">
        <param name="robot_name" value="$(arg robot_name)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        <param name="urdf_url" value="$(arg urdf_url)"/>
        <param name="controller_name" value="orca_controller"/>
        <param name="robot_compensates_gravity" value="$(arg robot_compensates_gravity)"/>
    </node>

    <node name="rviz_task_controller" pkg="orca_ros" type="rviz_task_controller" output="screen">
        <param name="robot_name" value="$(arg robot_name)"/>
        <param name="controller_name" value="orca_controller"/>
        <param name="task_name" value="$(arg task_name)"/>
        <param name="interactive_marker_frame_id" value="$(arg interactive_marker_frame_id)"/>
    </node>

    <arg name="rviz_config" default="$(find orca_ros)/config/simple_lwr_model.rviz"/>
    <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz" args="-d $(arg rviz_config)"/>

    <group ns="/orca">
        <group ns="$(arg robot_name)">

            <param name="robot_description" command="$(find xacro)/xacro '$(arg urdf_url)' --inorder"/>
            <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen"/>
        </group>
    </group>
</launch>
